javav2/usecases/creating_scheduled_events/src/main/java/com/aws/example/ScanEmployees.java

```diff
@@ -0,0 +1,130 @@
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.DynamoDbException;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.PublishRequest;
+import software.amazon.awssdk.services.sns.model.SnsException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+/*
+    Sends a text message to any employee that reached the one year anniversary mark
+ */
+
+public class ScanEmployees {
+
+      public Boolean sendEmployeMessage() {
+
+        Boolean send = false;
+        String myDate = getDate();
+
+        Region region = Region.US_WEST_2;
+        DynamoDbClient ddb = DynamoDbClient.builder()
+                .region(region)
+                .build();
+
+        // Create a DynamoDbEnhancedClient and use the DynamoDbClient object
+        DynamoDbEnhancedClient enhancedClient = DynamoDbEnhancedClient.builder()
+                .dynamoDbClient(ddb)
+                .build();
+
+        // Create a DynamoDbTable object based on Employee
+        DynamoDbTable<Employee> table = enhancedClient.table("Employee", TableSchema.fromBean(Employee.class));
+
+        try {
+            AttributeValue attVal = AttributeValue.builder()
+                    .s(myDate)
+                    .build();
+
+            // Get only items in the Employee table that match the date
+            Map<String, AttributeValue> myMap = new HashMap<>();
+            myMap.put(":val1", attVal);
+
+            Map<String, String> myExMap = new HashMap<>();
+            myExMap.put("#startDate", "startDate");
+
+            Expression expression = Expression.builder()
+                    .expressionValues(myMap)
+                    .expressionNames(myExMap)
+                    .expression("#startDate = :val1")
+                    .build();
+
+            ScanEnhancedRequest enhancedRequest = ScanEnhancedRequest.builder()
+                    .filterExpression(expression)
+                    .limit(15) // you can increase this value
+                    .build();
+
+            // Get items in the Employee table
+            Iterator<Employee> employees = table.scan(enhancedRequest).items().iterator();
+
+            while (employees.hasNext()) {
+                Employee employee = employees.next();
+                String first = employee.getFirst();
+                String phone = employee.getPhone();
+
+                // Send an anniversary message!
+                sentTextMessage(first, phone);
+                send = true;
+            }
+        } catch (DynamoDbException e) {
+            System.err.println(e.getMessage());
+            System.exit(1);
+        }
+        return send;
+    }
+
+
+    // Use the Amazon SNS Service to send a text message

```

Comment:
"Amazon SNS" -> "Amazon Simple Notification Service (Amazon SNS)" on first mention in a code file.

Reply:
fixed